bytecode/src/execute.py

```diff
@@ -0,0 +1,231 @@
+        return
+
+    def execute_return(self, code):
+        assert code.code == Mnemonics.RETURN
+        if code.value == None:
+            return NilFlag()
+
+        value = self.get(code.value)
+        if value == None:
+            return NilFlag()
+        else:
+            return value
+    # comparisons
+    equals_closure = BinaryEvaluator(lambda a, b: a == b, Mnemonics.EQUALS)
+    logical_and_closure = BinaryEvaluator(lambda a, b: a and b, Mnemonics.LOGICAL_AND)
+    logical_or_closure = BinaryEvaluator(lambda a, b: a or b, Mnemonics.LOGICAL_OR)
+    less_than_closure = BinaryEvaluator(lambda a, b: a < b, Mnemonics.LESS_THAN)
+    less_than_equals_closure = BinaryEvaluator(lambda a, b: a <= b, Mnemonics.LESS_THAN_EQUALS)
+    greater_than_closure = BinaryEvaluator(lambda a, b: a > b, Mnemonics.GREATER_THAN)
+    greater_than_equals_closure = BinaryEvaluator(lambda a, b: a >= b, Mnemonics.GREATER_THAN_EQUALS)
+
+    # arithmetic
+    addition_closure       = BinaryEvaluator(lambda a, b: a + b, Mnemonics.ADDITION)
+    subtraction_closure    = BinaryEvaluator(lambda a, b: a - b, Mnemonics.SUBTRACTION)
+    division_closure       = BinaryEvaluator(lambda a, b: a / b, Mnemonics.DIVISION)
+    multiplication_closure = BinaryEvaluator(lambda a, b: a * b, Mnemonics.MULTIPLICATION)
+
+    # -----
+
+    execute_byte : dict[Mnemonics, Callable[[Any, Bytecode], Any] ] = {
+        Mnemonics.NIL_CODE        : lambda self, code: None,
+        # general 
+        Mnemonics.NOOP            : lambda self, code: None,
+        Mnemonics.CREATE          : execute_create,
+        Mnemonics.PUSH_IDENTIFIER : execute_push,
+        Mnemonics.PUSH_VALUE      : execute_push,
+
+        # jumps
+        Mnemonics.JUMP            : execute_jump,
+        Mnemonics.JUMP_IF_TRUE    : execute_jump_if_true,
+        Mnemonics.JUMP_IF_FALSE   : execute_jump_if_false,
+
+        # function
+        Mnemonics.FUNCTION        : execute_function,
+        Mnemonics.CALL            : execute_call, 
+        Mnemonics.RETURN          : execute_return,
+
+        # comparisons
+        Mnemonics.EQUALS               : equals_closure.evaluate,
+        Mnemonics.LESS_THAN            : less_than_closure.evaluate,
+        Mnemonics.LESS_THAN_EQUALS     : less_than_equals_closure.evaluate,
+        Mnemonics.GREATER_THAN         : greater_than_closure.evaluate,
+        Mnemonics.GREATER_THAN_EQUALS  : greater_than_equals_closure.evaluate,
+        Mnemonics.LOGICAL_AND          : logical_and_closure.evaluate,
+        Mnemonics.LOGICAL_OR           : logical_or_closure.evaluate,
+
+        # arithmetic
+        Mnemonics.ADDITION       : addition_closure.evaluate,
+        Mnemonics.SUBTRACTION    : subtraction_closure.evaluate,
+        Mnemonics.DIVISION       : division_closure.evaluate,
+        Mnemonics.MULTIPLICATION : multiplication_closure.evaluate,
+    }
+    assert len(execute_byte.keys()) == len(Mnemonics)
```

Comment:
what is this for?

Reply:
I wanted something like a switch in python and the assertion is just to make sure it's exhaustive

Reply:
Yeah I wanted to use that but there isnt a stable version for apple silicon and I didnt feel like building from source